cgi-bin/LJ/Talk.pm

```diff
@@ -2549,6 +2545,292 @@ sub enter_imported_comment {
+    return $mlerr->('talk.error.purged') if $journalu->is_expunged;
+    return $err->("Account is locked, unable to post or edit a comment.") if $journalu->is_locked;
+    # can ANYONE comment?
+    return $mlerr->("$tp_d.error.nocomments") if $entry->comments_disabled;
+    # no replying to suspended entries, even by entry poster
+    return $mlerr->("$tp_d.error.noreply_suspended") if $entry->is_suspended;
+    # check max comments (unless editing existing comment)
+    if ( ! $content->{editid} && over_maxcomments( $journalu, $entry->jitemid ) ) {
+        return $mlerr->( "$tp_d.error.maxcomments" );
+    }
+    # If replying to a comment, it's gotta exist. (Hold onto it, we'll want it later.)
+    my $parenttalkid = ($content->{parenttalkid} || $content->{replyto} || 0) + 0;
+    my $parpost;
+    if ($parenttalkid) {
+        my $dbcr = LJ::get_cluster_def_reader($journalu);
+        return $mlerr->('error.nodb') unless $dbcr; # tbh we got bigger problems at this point.
+        $parpost = LJ::Talk::get_talk2_row( $dbcr, $journalu->{userid}, $parenttalkid );
+        unless ($parpost) {
+            return $mlerr->("$tp_d.error.noparent");
+        }
+    }
+    # no replying to frozen comments
+    my $parent_state = $parpost->{state} // '';
+    return $mlerr->( "$tp_d.error.noreply_frozen" ) if $parent_state eq 'F';
+
+    # Next: Permissions checks! Easily solved, just become someone else.
+
+    # (For switching between variant error messages.)
+    my $iscomm = $journalu->is_community ? '.comm' : '';
+
+    if ( $commenter_is_user ) {
+
+        # test accounts can only comment on other test accounts.
+        if (   ( grep { $commenter->user eq $_ } @LJ::TESTACCTS )
+            && !( grep { $journalu->user eq $_ } @LJ::TESTACCTS )
+            && !$LJ::IS_DEV_SERVER )
+        {
+            $mlerr->("$tp_d.error.testacct");
+        }
+
+        # Ban check for journal and entry:
+        if ($journalu->has_banned($commenter)) {
+            $mlerr->("$tp_d.error.banned$iscomm")
+        } else {
+            # comm hasn't banned you, but maybe this poster did
+            $mlerr->("$tp_d.error.banned.entryowner") if $entry->poster->has_banned($commenter);
+        }
+        # Ban check for parent comment:
+        my $parentu = LJ::load_userid( $parpost->{posterid} );
+        $mlerr->("$tp_d.error.banned.reply")
+            if defined $parentu && $parentu->has_banned($commenter);
+
+        # they down with unvalidated OpenIDs?
+        if ( $journalu->does_not_allow_comments_from_unconfirmed_openid($commenter) )
+        {
+            $mlerr->(
+                "$tp_d.error.noopenidpost",
+                {
+                    aopts1 => "href='$LJ::SITEROOT/changeemail'",
+                    aopts2 => "href='$LJ::SITEROOT/register'"
+                }
+            );
+        }
+
+        # No one's down with unvalidated site users.
+        # (FYI, nothing in -free or -nonfree ever calls that hook. -NF)
```

Comment:
In the future, I've been ripping out any hook that we aren't calling. Simplifies reasoning about the code, and we can always re-add them in the future if needed. Most hooks were done to add ads and other "business options" with LJ, which isn't really applicable to us. Or other weird features.

Reply:
Did we have an issue open at one point for looking at all the checks for hooks and ripping out all the ones that were never defined anywhere? Or was that just something we talked about doing?

Reply:
I think it just went past in conversation. Anyway, I'm 💯  for amending this PR to remove any more old garbage you happen to spot!

Reply:
@kareila don't think there's an issue open that I'm aware of, I've just been doing it from time to time in the name of code machete.